pkg/lockfile/parse-conan-lock.go

```diff
@@ -0,0 +1,182 @@
+package lockfile
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+)
+
+type ConanReference struct {
+	Name            string
+	Version         string
+	Username        string
+	Channel         string
+	RecipeRevision  string
+	PackageID       string
+	PackageRevision string
+	TimeStamp       string
+}
+
+type ConanGraphNode struct {
+	Pref      string `json:"pref"`
+	Ref       string `json:"ref"`
+	Options   string `json:"options"`
+	PackageId string `json:"package_id"`
+	Prev      string `json:"prev"`
+	Path      string `json:"path"`
+	Context   string `json:"context"`
+}
+
+type ConanGraphLock struct {
+	Nodes map[string]ConanGraphNode `json:"nodes"`
+}
+
+type ConanLockFile struct {
+	Version string `json:"version"`
+	// conan v0.4- lockfiles use "graph_lock", "profile_host" and "profile_build"
+	GraphLock    ConanGraphLock `json:"graph_lock,omitempty"`
+	ProfileHost  string         `json:"profile_host,omitempty"`
+	ProfileBuild string         `json:"profile_build,omitempty"`
+	// conan v0.5+ lockfiles use "requires", "build_requires" and "python_requires"
+	Requires       []string `json:"requires,omitempty"`
+	BuildRequires  []string `json:"build_requires,omitempty"`
+	PythonRequires []string `json:"python_requires,omitempty"`
+}
+
+// TODO this is tentative and subject to change depending on the OSV schema
+const ConanEcosystem Ecosystem = "ConanCenter"
+
+func parseConanRenference(ref string) ConanReference {
+	// very flexible format name/version[@username[/channel]][#rrev][:pkgid[#prev]][%timestamp]
+	var reference ConanReference
+
+	parts := strings.SplitN(ref, "%", 2)
+	if len(parts) == 2 {
+		ref = parts[0]
+		reference.TimeStamp = parts[1]
+	}
+
+	parts = strings.SplitN(ref, ":", 2)
+	if len(parts) == 2 {
+		ref = parts[0]
+		parts = strings.SplitN(parts[1], "#", 2)
+		reference.PackageID = parts[0]
+		if len(parts) == 2 {
+			reference.PackageRevision = parts[1]
+		}
+	}
+
+	parts = strings.SplitN(ref, "#", 2)
+	if len(parts) == 2 {
+		ref = parts[0]
+		reference.RecipeRevision = parts[1]
+	}
+
+	parts = strings.SplitN(ref, "@", 2)
+	if len(parts) == 2 {
+		ref = parts[0]
+		UsernameChannel := parts[1]
+
+		parts = strings.SplitN(UsernameChannel, "/", 2)
+		reference.Username = parts[0]
+		if len(parts) == 2 {
+			reference.Channel = parts[1]
+		}
+	}
+
+	parts = strings.SplitN(ref, "/", 2)
+	if len(parts) == 2 {
+		reference.Name = parts[0]
+		reference.Version = parts[1]
+	} else {
+		reference.Version = ref
+	}
```

Comment:
I suspect I'm about to open a can of worms with this (which I missed in my first review, sorry about that), but this looks to me like it's possible for a package to not have a name - is that true? technically the `osv.dev` API allows that if you have a commit, but it's not really expected elsewhere.

it also looks like none of your tests are actually covering this line (unless I've missed something)

Reply:
All the official packages that are open-source that could possibly be in public databases with CVE will have a name.

It's technically possible for someone to have private packages but I would safe those are safe to skip.

Perhaps a warning or just skipping the package is an option?

Reply:
I've just got some responses:
```
It is possible to have self.version defined for a consumer, but not have self.name defined
In conan export or conan create it should never happen, I understand that it has reached the lockfile because it is a consumer
Well, not having to define a name, a conanfile.txt has neither a name nor a version, a conanfile.py also works fine without a name or version
But if you want to give it a version, you can.
```
so, I think they might theoretically appear in a lockfile, but they should be completely safe to skip.
I'll try to add some test case, if I can generate such a lockfile, otherwise, I'll just skip them in code (if Name is missing or empty).

Reply:
I've added code to skip packages with no name